server/sonar-server/src/main/java/org/sonar/server/computation/step/ValidateProjectStep.java

```diff
@@ -147,10 +147,10 @@ private void validateProjectKey(Optional<ComponentDto> baseProject, String rawPr
 
     private void validateAnalysisDate(Optional<ComponentDto> baseProject) {
       if (baseProject.isPresent()) {
-        SnapshotDto snapshotDto = dbClient.snapshotDao().selectLastSnapshotByComponentUuid(session, baseProject.get().uuid());
+        java.util.Optional<SnapshotDto> snapshotDto = dbClient.snapshotDao().selectLastSnapshotByRootComponentUuid(session, baseProject.get().uuid());
```

Comment:
should method `SnapshotDao.selectLastSnapshotByRootComponentUuid` be renamed after the change of cardinality?